plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/BaseOracleConnectorTest.java

```diff
@@ -191,6 +195,119 @@ public void testTimestampOutOfPrecisionRounded()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testCastProjectionPushdown()
```

Comment:
extract it as a `TestCast` class and inject to `BaseOracleConnectorTest` as `@Nested`